### ISSUE TITLE
[spec] Automating Case Storage assignment to judges: 3rd draft

### ISSUE THREAD

**Issue by laurjpeterson** [department-of-veterans-affairs/caseflow#4985]
# Context

See https://github.com/department-of-veterans-affairs/appeals-design-research/issues/645
Earlier draft of this spec https://github.com/department-of-veterans-affairs/caseflow/issues/4954

# Overview

We'll develop a Case Storage auto-assignment model that:
 - removes any need for a Case Storage staff member to push cases to judges and eliminates lag time
 - allows the Board to monitor and adjust the throughput of different types of cases so it can remain compliant & prevent logjams
 - will leave room to add three types of AMA appeals
 - will allow us to add specialty case teams. Specialty cases can be distributed directly to attorneys, bypassing judges. This is out of scope for this spec.

# Implementation

## Chris's drafty draft proposal for auto-assignment logic:

* Each judge has a set **individual assignment batch size**. This could be a uniform number for everyone, or could be determined by the size of the judge's team or other factors.

* The sum of all judges' assignment batch sizes is the **total assignment batch size**. Let's say there are 90 judges with an average assignment batch size of 40 cases. This would give us a total assignment batch size of 3,600.

* We will divide this assignment batch size among the dockets according to logic to be determined. Each docket will return cases according to its own logic. Initially, the only docket is the legacy docket, so the **total docket batch size** for the legacy docket will be 100% of the assignment batch size, or 3,600. Subsequent bullets only apply to the legacy docket.

* Find the number of AOD cases ready to be assigned. Let's say there are 600 AOD cases. Subtracting this from the total docket batch size will give us the **net docket range**, 3,000. This number cannot be less than zero.

* Retrieve the (net docket range)-oldest non-AOD cases that are ready to be assigned. In combination with the AOD cases that are ready to be assigned, these are our **assignable cases**. (If there are more AOD cases than the total docket batch size, we only use the (total docket batch size)-oldest AOD cases.)

* Divide the assignable cases into **hearing cases** and **genpop cases**. Divide the hearing cases into cases that are assignable to this judge (the judge who has pushed the "I can haz appealz?" button), **own hearing cases** and cases that are not, **other hearing cases**. (A case with a panel decision would be assignable to multiple judges.)

* Subtract the number of hearing cases, including AOD and non-AOD, from the total docket batch size to arrive at the **net non-hearing docket batch size**. Let's say there are 1,600 hearing cases; that would give us 2,000.

* Divide the total number of non-hearing AOD cases by the net non-hearing docket batch size to arrive at the **genpop AOD rate**. Let's say there were 400 non-hearing AOD cases; that would give us a rate of 20%.

* Now we have sufficient information to retrieve a number of cases equal to the individual assignment batch size, in this example 40. We do so in the following order:

  * AOD cases from the own hearing cases, in time pending order order. Let's say there were 2 such cases; leaving us with 38 slots.
  * Other cases from the own hearing cases, in docket order. Let's say there were 18 such cases, leaving us with 20 slots.
  * If there are still slots remaining, multiply the number of slots by the **genpop AOD rate. The 20 remaining slots, times 20%, gives us 4. So we'll assign 4 genpop AOD cases, in time pending order, leaving us with 16 slots.
  * Any remaining slots are given to non-AOD genpop cases. So we'll assign 16 genpop non-AOD cases, in docket order.
  * If there are not enough non-hearing cases to fill the remaining slots, we'll look outside the assignable cases we've retrieved and find the next n cases that are either own hearing cases or non-hearing cases, up to either the total number of remaining slots or until there are no more cases older than the Board's docket month.

## Monitoring

To validate our model is behaving as we intend and so we can watch for any needed adjustments, we'll need to track the flow of all types of cases to judges separately. However, in the end, the top-line metric we care about is decision output, not case assignment. We'll want to also ensure descision output by case type is surfaced and easily available. We might opt to do this in Datadog at first, but ideally we'd make this easy to view for BVA employees as well, and ensure Tableau or a business metric tool can surface necessary data.

## Manual assignment

It's likely that there will be assignment hiccups. When we initially roll out case storage auto-assignment, we should ensure we can manually tweak case assignments as needed. At first, this can just be through a method exposed to the Rails console that a system admin engineer can run. This manual method must be able to support switching cases from judge to judge.

## User Interface: Self-service button

When a judge loads Caseflow Queue and views the appeals assigned to them, we could present the judge a "Fetch more from Case Storage" button. Upon clicking a button, we'd assign a batch of case storage cases to the judge, and they'd appear in the Queue list view immediately after assignment is complete.

#### Goal: Judges should understand the methodology behind automatic case allocation
  - The button would require minimal training and explanation
#### Goal: Cases should not be allocated to attorneys and judges on PTO/leave
  - Judges are likely in the best position to know who is on PTO/leave, and they won't request cases for themselves while they're on leave
#### Goal: Cases are only allocated to active staff 
  - Cases will not be assigned to judges unless they click a button, and judges will know not to assign cases to inactive attorneys
#### Goal: Cases should not sit on judges plates to assign or to sign for long times
  - A "Fetch more from Case Storage" button could be disabled until after a judge has assigned all existing cases 

# Test Plan

Using FACOLS, we'll be able to thoroughly test the Case Storage assignment to judge logic locally. 

# Rollout Plan

We can develop this in parallel with the judge queue, but we shouldn't roll it out until we roll out the judge queue. We could test this process change with one judge at first by putting it behind a feature toggle.

**Comment by laurjpeterson** [comment]
@cmgiven @amprokop 

Today, Senior Counsel request gen pop cases from case storage so they can assign overtime cases to attorneys in their chief group. This is currently not a duty judges are authorized to do, and it would be a larger Board leadership decision to ask judges, instead of senior counsel, to do this.

Some Senior Counsel (Jebby) also request AOD and CAVC cases to distribute evenly among their chief group. We are eliminating their need to do this with the logic listed above for automatic case assignment, and I've been questioning the efficiency of senior counsel and others doing so much active "case management", so I initially was thinking we could do the same with overtime cases

But posing the question to you both - what if we added this senior counsel caveat to our automatic case assignment logic?

## Proposed requirements
- Senior Counsel can request cases from case storage in Caseflow, just like we are proposing for judges.
- When Senior Counsel requests cases, they are assigned x number of gen pop cases - defined as non-hearing or retired judge hearing, non-AOD cases.

**Comment by cmgiven** [comment]
That would be pretty easily incorporated. I would suggest just not changing the total batch size used for calculating the AOD rate, as it seems like this will be more sporadic.

The other option, though, would be to enable the lawyer to request the next available genpop case directly. Depending on what we want, that could either be never AOD, the same probability of AOD as judge-requested cases, or always AOD if genpop AOD is available.

**Comment by laurjpeterson** [comment]
Interesting thought. Given the approval process for overtime cases, I'm not sure the Board would buy into attorneys requesting overtime cases directly. @nicholasholtz what do you think?

**Comment by laurjpeterson** [comment]
potential wrinkle - chief judges will have to request cases because they have to sign 5 cases per week (compared to the VLJ quota of 20 cases per week), but they don't have any attorneys. how will we define their batch size?

**Comment by cmgiven** [comment]
We can handle that. We'll just define their batch size at 5, and add (number of chiefs * 5) to the total batch size. Or we can do 10 if they prefer more of a biweekly schedule.

**Comment by laurjpeterson** [comment]
## Auto-case distribution section of 90 day report

The Board remains committed to an average processing time goal of 365 days for direct docket appeals – appeals received under the new framework that do not include a request for a hearing or an opportunity to submit additional evidence. To accomplish this goal, Digital Service is using a technique called Discrete Event Simulation (DES) to model the Board's dockets and to test and optimize the automatic case distribution algorithm prior to implementation. The outputs of this simulation measure its performance according to identified business objectives. The Board's 365-day average processing time goal for cases on the direct docket is a "constraint" on the algorithm, meaning that direct docket cases must be distributed at a rate that allows the Board to meet its goal. Each case on the direct docket will have a "target distribution date," which will take into account both the date on which the case must be decided in order to meet the timeliness goal (365 days from the date of the NOD), and the average amount of time it takes to issue a decision from the date that the case is assigned to a Veterans Law Judge. The distribution rate of cases on the direct docket will be determined in part by the total number of direct docket cases with a target distribution date close to the current date.
Cases on the legacy, evidence, and hearing dockets are distributed at a rate proportionate to the number of cases on each docket. A natural bias for legacy cases will be created, which ensures that the legacy cases in the Board’s working docket range will be worked prior to those in the hearing or evidence lanes of the new system. The case distribution algorithm will take into account pending remands and historical receipt rates of substantive appeals in calculating the legacy inventory to ensure those appeals are not undercounted. Initially, the Board's inventory of legacy appeals will be larger than its inventory of new system appeals. Therefore, the Board will work a higher percentage of legacy appeals in the first year following the effective date.
The case distribution algorithm must also take into account certain statutory requirements that affect the order in which the Board may decide appeals. Legacy appeals must be decided in order according to docket date, which is based on the date of receipt of a substantive appeal. Legacy appeals retain their original docket date upon return from the agency of original jurisdiction after a remand by the Board. In contrast, appeals under the new framework must be decided in the order they are received on each docket. Under both the legacy and new framework, cases may be advanced on their docket due to the Veteran's age, illness, financial hardship, or other reasons. Cases that have been remanded to the Board by the Court of Appeals for Veterans' Claims must be expedited. Moreover, legacy appeals in which the Veteran testified before a Veterans Law Judge must be decided by the same Veterans Law Judge who presided over the hearing, while no such requirement exists for appeals in the new system.
As a policy matter, the Board may apply additional constraints to the case distribution algorithm. For example, distribution of direct docket cases may be accelerated during the first few months following the effective date. Otherwise, these cases would not be distributed until their target distribution date, which would be nearly a year after the effective date. Placing timeliness contraints on the evidence and hearing dockets would restrict the algorithm's ability to distribute cases proportionate to inventory, prioritize cases based on statutory requirements, and ensure that the direct docket goal is met. If the Board imposed timeliness constraints on all three dockets, there is significant risk that it would be unable to satisfy all of those constraints simultaneously. Initially, the Board anticipates that cases in the evidence and hearing dockets will take longer than those in the direct docket, as a result of the prioritization of legacy and direct docket appeals.
Rather than setting goals for the evidence and hearing docket timeliness prior to implementation, the Board will publish projected wait times for each docket after implementation, based on early results, with regular updates. By providing transparent, realistic timeliness estimates for the other two Board dockets, VA empowers Veterans to make informed choices as to which Board option best suits their individual needs. User research conducted during BEAAM, discussed in more detail in section 4, supports this approach. In particular, Veterans who were interviewed reported that they have limited confidence in timeliness estimates that do not reflect their experiences. Veterans expressed a preference for dynamic time estimates for each option which show a range of durations, rather than an average. Moreover, Veteran representatives expressed that they would cease to recommend an option if it proved unreliable, for example, if a stated timeliness goal was not realistic. Transparency not only assists Veterans in making informed choices, it also allows Veteran representatives to more effectively inform and advocate for their clients, for example, by providing a Veteran facing homelessness with an accurate comparison between wait times on the Board’s hearing and direct dockets.
